pkgs/build-support/fetchgit/default.nix

```diff
@@ -12,7 +12,22 @@
       else "";
   in "${if matched == null then base else builtins.head matched}${appendShort}";
 in
-{ url, rev ? "HEAD", md5 ? "", sha256 ? "", leaveDotGit ? deepClone
+{ url
+, rev ? "HEAD"
+
+, # SRI hash.
+  hash ? ""
```

Comment:
This should be called `narHash` for consistency with `builtins.fetchX` (https://github.com/NixOS/nix/commit/d4df99a3349cf2228a8ee78dea320afef86eb3ba).

Reply:
Should I rename it in https://github.com/NixOS/nixpkgs/commit/267c8d6b2fea05bc811c8e2c2f4529b1436eeb9a as well?

Reply:
Yes, probably. We could also rename everything to `hash`, but that's ambiguous in functions that take both a file and a NAR hash...

Reply:
Does it make sense to call it `narHash` if it can also be used for `outputHashMode = "flat"`? If I understand the [docs](https://nixos.org/nix/manual/#fixed-output-drvs) correctly, hash of the NAR is only used for FOD when using `recursive` output hash mode.

Reply:
> functions that take both a file and a NAR hash

I am not sure what you mean.

Reply:
Alternately, we could use `outputHash` directly, as that is what gets passed to Nix in the end.

Reply:
Since we are using SRI hashes, using `integrity` attribute name like in HTML might actually fit even better.